### PR TITLE
feat(icons): clear icons Cu 1x73an0

### PR DESCRIPTION
**Pull request by snqb**
remove fontawesome, one source of truth for icons

This is basically an addition to @giardiv 's work in the https://github.com/logicalclocks/quartz/pull/180

BREAKING CHANGE: Icons and Icon components are removed

### CODE REVIEW

File: src/components/button/button.styles.ts [hunk 1]
```diff
@@ -26,6 +26,10 @@ export default {
 
   '> span': {
     mr: '8px',
+    display: 'flex',
+    height: '15px',
```
Comment (snqb) [inline src/components/button/button.styles.ts:30]: this was added by @giardiv

Comment (snqb) [reply src/components/button/button.styles.ts:30]: and all of other CSS things
I was mostly dealing with other things :D

File: src/components/button/utils.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    case 'ghost':
+      return 'primary';
+    case 'ghost':
+      return 'primary';
+    case 'alert':
+      return 'labels.red';
+    default:
+      return 'primary';
+  }
+}
```
Comment (esonhey) [inline src/components/button/utils.ts]: neat: would be more beautiful if you use objects here as well as you did in other places

File: src/components/icon/GetIcon.tsx [hunk 3]
```diff
@@ -1,26 +1,23 @@
...
 
 export interface GetIconProps extends Omit<BoxProps, 'css' | 'color'> {
   icon: IconName;
-  color?:
-    | Exclude<keyof IThemeColors, 'labels'>
-    | `labels.${keyof IThemeColors['labels']}`;
+  color?: Colors;
   /** width of the svg, height will scale accordingly */
-  size?: keyof IThemeIconSizes;
+  size?: IconSizes;
```
Comment (esonhey) [inline src/components/icon/GetIcon.tsx]: nice extraction of Type to reuse!

File: src/components/typography/input-validation/utils.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { GetIcon, IconName } from '../../icon';
+
+const icons: Record<string, React.ReactElement> = {
+  success: <GetIcon icon={IconName.validate} color="primary" size="sm" />,
+  warning: <GetIcon icon={IconName.warning} color="labels.orange" size="sm" />,
+  neutral: <GetIcon icon={IconName.info} color="gray" size="sm" />,
+};
```
Comment (esonhey) [inline src/components/typography/input-validation/utils.tsx:8]: beautiful!

File: src/theme/types.ts [hunk 5]
```diff
@@ -29,6 +29,11 @@ export interface IThemeColors {
   };
 }
 
+type LabelColor = keyof IThemeColors['labels'];
+export type Colors = keyof IThemeColors | `labels.${LabelColor}`;
```
Comment (esonhey) [inline src/theme/types.ts]: neat: it's better to have singular `Color` type

File: src/components/callout/index.tsx [hunk 6]
```diff
@@ -10,6 +11,15 @@ export enum CalloutTypes {
   neutral = 'neutral',
 }
 
+const typeToColor = (type: CalloutTypes): Colors => {
+  return {
+    valid: 'primary' as const,
```
Comment (esonhey) [inline src/components/callout/index.tsx]: why would you need the assert it as const?

**Review by esonhey — APPROVED** [review]
Nice job! just left a few comments to make it more beautiful

would you move all the icons in hopsworks to quartz to keep them all in one place?
also would be good addition if you add searching feature over icons by their name

**Comment by snqb** [comment]
hey @ehsan-github 
thanks, I'll update things tomorrow

Search is a great idea, have been thinking about it for a while.
Guess it won't hurt adding in this PR

some of the stuff you commented was kinda straight-up copy-paste. I'll fix those, everything was straight to the point.

Also, yeah, hopsworks side is coming up next. Basically, I'm thinking there should be one source of truth: `GetIcon`. Everything else should be gone.

**Comment by snqb** [comment]
Added search

image
image

**Pull request merged**